src/id.rs

```diff
@@ -22,7 +22,6 @@ use NameType;
 // Note: name field is initially same as original_name, this should be later overwritten by
 // relocated name provided by the network using assign_relocated_name method
 // TODO (ben 2015-04-01) : implement order based on name
-#[derive(Clone)]
```

Comment:
I was planning on cloning the Id from the initial Routing object into the Routing core, as vaults acting on Routing need to sign messages as the RoutingNode; alternatively we need to have a separate Action:: that takes the elements of a message, but not a fully SignedMessage, and then finish construction in RoutingNode